### ISSUE TITLE
Problems with shutting down Golem on develop source

### ISSUE THREAD

**Issue by ederenn** [golemfactory/clay#4809]
## Description

**Golem Version**: 0.21.0+dev122.g9a56168

**Golem-Messages version** (leave empty if unsure): golem_messages Version: 3.13.0

**Electron version** (if used): 0.2.3

**OS** [e.g. Windows 10 Pro]: Linux

**Branch** (if launched from source): develop

**Mainnet/Testnet**: mainnet

**Priority label is set to the lowest by default. To setup higher priority please change the label**
_P0 label is set for Severity-Critical/Effort-easy
P1 label is set  for Severity-Critical/Effort-hard
P2 label is set for Severity-Low/ Effort-easy
P3 label is set for Severity-Low/Effort-hard_

**Description of the issue**:

Stopping Golem running with console window with ctrl c gives a critical

**Actual result**:

Golem is not fully shutting down. To have it shut i need to kill Golem and crossbar processes.

after ctrl c

```
ederenn@GOLEM2008-VM-UBNT18:~/project/golem$ ps aux | grep golem
ederenn    5930  0.1  3.4 1776544 570944 pts/3  Sl+  13:43   0:19 node /home/ederenn/project/golem-electron/node_modules/.bin/webpack-dev-server --mode development --hot --port 3002
ederenn    6300  0.0  0.0  12188  5216 pts/5    S+   14:35   0:00 golemcli -i
ederenn    7675  0.0  0.0  22756  1064 pts/7    S+   16:25   0:00 grep --color=auto golem
ederenn@GOLEM2008-VM-UBNT18:~/project/golem$ ps aux | grep crossbar
ederenn    5758  3.6  1.0 2795072 176984 pts/0  Sl+  13:42   5:53 crossbar-controller
ederenn    7677  0.0  0.0  22756  1056 pts/7    S+   16:25   0:00 grep --color=auto crossbar
ederenn@GOLEM2008-VM-UBNT18:~/project/golem$ 
```

## Steps To Reproduce
_Short description of steps to reproduce the behavior:_
e.g.
1. Run Golem from console on develop branch
2. Wait to have app running
3. Close it with ctrl c

## Expected behavior
_(What is the expected behavior and/or result in this scenario)_

## Logs and any additional context
Linux:
```
WARNING  [twisted                            ] Controller received SIGINT [signal=2]: shutting down node [shutdown_was_clean=True] ..
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out

CRITICAL [twisted                            ] Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out
WARNING  [twisted                            ] Controller received SIGINT [signal=2]: shutting down node [shutdown_was_clean=True] ..
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out

CRITICAL [twisted                            ] Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out
WARNING  [twisted                            ] Controller received SIGINT [signal=2]: shutting down node [shutdown_was_clean=True] ..
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out

CRITICAL [twisted                            ] Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1451, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "/home/ederenn/project/golem-env/lib/python3.6/site-packages/twisted/internet/base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 663, in quit
    self.stop()
  File "/home/ederenn/project/golem/golem/report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/project/golem/golem/client.py", line 363, in stop
    self.task_server.quit()
  File "/home/ederenn/project/golem/golem/task/taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "/home/ederenn/project/golem/golem/core/deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out
```

Windows:
```
INFO [actix_server::builder] SIGINT received, exiting
INFO [actix_server::builder] SIGINT received, exiting
WARNING  [twisted                            ] Controller received SIGINT [signal=2]: shutting down node [shutdown_was_clean=True] ..
INFO     [golem.client                       ] Shutting down ...
INFO     [golem.client                       ] Stopping network ...
INFO     [golem.network.concent.client       ] Waiting for received messages queue to empty
INFO     [golem.network.concent.client       ] <ConcentClientService(Thread-7, started daemon 8540)> stopped
INFO     [golem.core.processmonitor          ] Subprocess 3652 exited with code 0
INFO     [golem.resources                    ] hyperg 0.3.6 already started. addresses={'TCP': ('0.0.0.0', 3282)}
Unhandled Error
Traceback (most recent call last):
  File "C:\Users\ederenn\AppData\Local\Programs\Python\Python36\lib\asyncio\base_events.py", line 1431, in _run_once
    handle._run()
  File "C:\Users\ederenn\AppData\Local\Programs\Python\Python36\lib\asyncio\events.py", line 145, in _run
    self._callback(*self._args)
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem\golem\client.py", line 663, in quit
    self.stop()
  File "C:\Users\ederenn\Projects\golem\golem\report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem\golem\client.py", line 363, in stop
    self.task_server.quit()
  File "C:\Users\ederenn\Projects\golem\golem\task\taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "C:\Users\ederenn\Projects\golem\golem\core\deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out

CRITICAL [twisted                            ] Unhandled Error
Traceback (most recent call last):
  File "C:\Users\ederenn\AppData\Local\Programs\Python\Python36\lib\asyncio\base_events.py", line 1431, in _run_once
    handle._run()
  File "C:\Users\ederenn\AppData\Local\Programs\Python\Python36\lib\asyncio\events.py", line 145, in _run
    self._callback(*self._args)
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\asyncioreactor.py", line 290, in run
    f(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\base.py", line 721, in fireSystemEvent
    event.fireEvent()
--- <exception caught here> ---
  File "C:\Users\ederenn\Projects\golem-env\lib\site-packages\twisted\internet\base.py", line 428, in fireEvent
    result = callable(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem\golem\client.py", line 663, in quit
    self.stop()
  File "C:\Users\ederenn\Projects\golem\golem\report.py", line 173, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\ederenn\Projects\golem\golem\client.py", line 363, in stop
    self.task_server.quit()
  File "C:\Users\ederenn\Projects\golem\golem\task\taskserver.py", line 310, in quit
    sync_wait(defer_rtm_quit)
  File "C:\Users\ederenn\Projects\golem\golem\core\deferred.py", line 55, in sync_wait
    raise defer.TimeoutError("Command timed out")
twisted.internet.defer.TimeoutError: Command timed out

WARNING  [twisted                            ] session closed with reason wamp.close.transport_lost [()]
WARNING  [twisted                            ] Native worker connection closed uncleanly: A process has ended with a probable error condition: process ended with exit code 1.
```

might be related:
https://github.com/golemfactory/golem/issues/4768

## Proposed Solution?
_(Optional: What could be a solution for that issue)_

**Comment by etam** [comment]
resolved by #4842, but it was merged into `mwu/e2e-rtm` branch, not `b0.21`.

**Comment by mfranciszkiewicz** [comment]
The fix has been merged to `develop` and `b0.22`. @ZmijaWA Can we close this issue?
